packages/toaster/components/Toaster.tsx

```diff
@@ -15,124 +15,96 @@ interface ToasterProps {
   children?: Toast | Toast[];
 }
 
-class Toaster extends React.PureComponent<ToasterProps, { toasts: Toast[] }> {
-  public timeouts: number[] = [];
+const Toaster = ({ children }: ToasterProps) => {
+  const timeouts = React.useRef<number[]>([]);
+  const toastChildren = React.Children.toArray(children) as Toast[];
 
-  constructor(props) {
-    super(props);
+  const [toasts, setToasts] = React.useState<Toast[]>(toastChildren);
 
-    this.cloneToast = this.cloneToast.bind(this);
-    this.dismissToast = this.dismissToast.bind(this);
-    this.clearTimeouts = this.clearTimeouts.bind(this);
-    this.restartTimeouts = this.restartTimeouts.bind(this);
-    this.setTimer = this.setTimer.bind(this);
-
-    this.state = {
-      toasts: React.Children.toArray<Toast>(this.props.children).map(
-        this.cloneToast
-      )
-    };
-  }
+  const handleDismissToast = (id: ToastId = "") => {
+    const toastToDismiss = toasts.find(toast => id === toast.props.id);
+    if (toastToDismiss && toastToDismiss.props.onDismiss) {
+      toastToDismiss.props.onDismiss();
+    }
+    setToasts(toasts.filter(toast => id !== toast.props.id));
+  };
+
+  const restartTimeouts = () => {
```

Comment:
shouldn't this be memoized inside useCallback?

Reply:
I don't typically memoize unless I have a particular need for it or notice performance issues. Would useCallback here preserve some functionality that the public method on the class had here? https://github.com/dcos-labs/ui-kit/blob/main/packages/toaster/components/Toaster.tsx#L129

Reply:
okay looks like this would maintain the same referential integrity that would have existed in the class, so I'll wrap the functions around it https://stackoverflow.com/questions/60344321/how-to-convert-functional-component-using-hooks-to-class-component